wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java

```diff
@@ -204,18 +205,16 @@ public boolean writeHPCCFileData(byte []  data, String fileName, String targetLa
 
         int dataindex = 0;
         int limit = uploadchunksize <= 0 ? defaultUploadChunkSize : uploadchunksize;
-        long payloadsize = 0;
-        long bytesleft = data.length;
+        int payloadsize = 0;
+        int bytesleft = data.length;
         byte [] subdata = null;
 
         while (bytesleft > 0)
         {
             payloadsize = bytesleft >= limit ? limit : bytesleft;
-            log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
+            log.trace("Writing offset: "+dataindex+"\t size: "+(dataindex + payloadsize));
 
-            subdata = new byte [(int)payloadsize];
-            for (int i = 0; i < payloadsize ; i++, dataindex++)
-                subdata[i] = data[dataindex];
+            subdata = Arrays.copyOfRange(data, 0, payloadsize-1);
```

Comment:
It looks like we aren't changing the copy offset. I think this should be something like:
`
int end = dataindex + payloadsize-1;
subdata = Arrays.copyOfRange(data, dataindex, end);
dataindex += payloadsize;
`